lib/defender/aws_iot_defender_reporter_socket.c

```diff
@@ -0,0 +1,255 @@
+                {
+                    for( uint8_t i = 0; i < total; i++ )
+                    {
+                        AwsIotSerializerEncoderObject_t connectionMap = AWS_IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_MAP;
+
+                        /* open a map under "connections" */
+                        error = _AwsIotDefenderEncoder.openContainer( &connectionsArray,
+                                                                      &connectionMap,
+                                                                      hasLocalPort + hasRemoteAddr );
+
+                        /* add local port */
+                        if( hasLocalPort && !error )
+                        {
+                            error = _AwsIotDefenderEncoder.appendKeyValue( &connectionMap, _LOCAL_PORT_TAG,
+                                                                           AwsIotSerializer_ScalarSignedInt( pConnections[ i ].localPort ) );
+                        }
+
+                        /* add remote address */
+                        if( hasRemoteAddr && !error )
+                        {
+                            SOCKETS_inet_ntoa( pConnections[ i ].remoteIP, remoteAddr );
+                            sprintf( remoteAddr, "%s:%d", remoteAddr, pConnections[ i ].remotePort );
+
+                            error = _AwsIotDefenderEncoder.appendKeyValue( &connectionMap, _REMOTE_ADDR_TAG,
+                                                                           AwsIotSerializer_ScalarTextString( pRemoteAddr ) );
+                        }
+
+                        if( !error )
+                        {
+                            error = _AwsIotDefenderEncoder.closeContainer( &connectionsArray, &connectionMap );
+                        }
+                    }
+                }
+
+                if( !error )
+                {
+                    error = _AwsIotDefenderEncoder.closeContainer( &establishedMap, &connectionsArray );
+                }
+
+                vPortFree( pConnections );
+            }
+            else
+            {
+                error = AWS_IOT_SERIALIZER_OUT_OF_MEMORY;
+            }
+        }
+
+        if( hasTotal && !error )
+        {
+            error = _AwsIotDefenderEncoder.appendKeyValue( &establishedMap,
+                                                           _TOTAL_TAG,
+                                                           AwsIotSerializer_ScalarSignedInt( total ) );
+        }
+
+        if( !error )
+        {
+            error = _AwsIotDefenderEncoder.closeContainer( &tcpConnectionMap, &establishedMap );
+        }
+    }
+
+    if( !error )
+    {
+        error = _AwsIotDefenderEncoder.closeContainer( pEncoderObject, &tcpConnectionMap );
+    }
+
+    return error;
+}
+
+/*-----------------------------------------------------------*/
+
+AwsIotSerializerError_t AwsIotDefenderInternal_GetLisTcpMetrics( AwsIotSerializerEncoderObject_t * pEncoderObject )
```

Comment:
If this function is only called in this file then it should be called
_GetLisTcpMetrics...

what does LIs means? it is not obvious, so consider changing name...

Reply:
This function is implemented in this file but used by "collector" to aggregate report.

It should be "Listening TCP Ports". I will change the name.